src/clj/puppetlabs/services/jruby/jruby_puppet_schemas.clj

```diff
@@ -108,7 +115,8 @@
    :borrow-timeout schema/Int
    :max-active-instances schema/Int
    :max-requests-per-instance schema/Int
-   :use-legacy-auth-conf schema/Bool})
+   :use-legacy-auth-conf schema/Bool
+   (schema/optional-key :environment-class-cache-enabled) schema/Bool})
```

Comment:
@cprice404 I definitely agree that that's the best pattern to follow and would be the ideal here if we had time. Do we have tickets anywhere about the work that will be required for splitting out the jruby config - seems like something that would be good to add there, or just create another ticket?